### PR TITLE
school library: set up associations

### PR DESCRIPTION
**Pull request by Alexr16**
### In this PR:
- Created a class Classroom.
- Created the `has-many/belongs-to` relationship between Classroom and Student.
- Created a class Book.
- Created a class Rental.
- Created the `many-to-many` relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (TedLivist) [inline book.rb]: - [ ] The `:rentals` property here should have a getter and setter, not just a setter. Kindly fix that.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (TedLivist) [inline book.rb]: - [ ] We need to set up the `has-many-through` relationship between the **Book** class and the **Person**, using the **Rental** class as an intermediate. Kindly do that.

  ## SUGGESTION
  In order to do that we can set up an `add_rental` method here that instantiates a rental object, like so:
  ```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
  ```

File: person.rb [hunk 3]
```diff
@@ -1,17 +1,20 @@
 require_relative 'nameable'
 require_relative 'capitalize_decorator'
 require_relative 'trimmer_decorator'
+require_relative 'rental'
+require_relative 'book'
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (TedLivist) [inline person.rb]: - [ ] The `rentals` variable should have `attr_accessor`, not `attr_reader`. Kindly do that.

File: person.rb [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
 
   def initialize(age, name = 'unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (TedLivist) [inline person.rb]: - [ ] Kindly add the method for setting up the `many-to-many` relationship as demonstrated earlier.

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (TedLivist) [inline rental.rb]: - [ ] Kindly make assign the `attr_accessor` to the `book` and `person` instance variables here.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Alexr16,

Good job so far!
✔️ Set up of association between classroom and students
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] I think you meant `cd OOP-school-library` here. Kindly fix that.

  alex-cd

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 5 above)
Comment (Alexr16) [reply rental.rb]: Hi @TedLivist, thanks for your review. The required changes were applied.

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @Alexr16 , 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**